File: C/1_02.c
#include <stdio.h>

// Find the last but one element of a list

#define MAX_LIST 10000

int lastElement(int list[])
{
	int i = 0;
	while (i < MAX_LIST && list[i] != '\0')
	{
		i ++;
	}
	return list[i - 2];  // -2
}

int main(void)
{
	int list[MAX_LIST] = {1, 2, 3, 4, 5, 6};
	int maxLst = lastElement(list);
	printf("last element = %d", maxLst);
}

File: ocaml/1_02.ml
let rec last_but_one l = match l with
  | lbo :: le :: [] -> lbo
  | _ :: tl  -> last_but_one tl
  | _ -> failwith "Failure, could not get last but one element of the list"


let () = 
  let a = 1 :: 2 :: 3 ::[] and b = 'y' :: 'z' :: [] in
  Printf.printf "last but one [1; 2; 3] = %d \nlast but one ['y'; 'z'] = %c\n" (last_but_one a) (last_but_one b);